### PR TITLE
add chart challenge code for drought duration plot

### PR DESCRIPTION
**Pull request by cnell-usgs**
The purpose of this PR is to level up and add some initial scripts to dig into the drought properties. It takes around 15 min to run this pipeline to get the chart at the end.

This is adding Scott's functions to process and plot the drought duration data in the unique "Stripswarm" chart. https://github.com/USGS-VIZLAB/chart-challenge-22/tree/main/19_shamshaw I'm applying this to the drought proporties data, for now using the `jd_7d` data and a limited timeseries to draw the chart, which looks like this:
duration_national

I've also added a target and to the metadata the CASC regions for an initial look at regional patterns.

### CODE REVIEW

File: _targets.R [hunk 1]
```diff
@@ -7,7 +7,14 @@ tar_option_set(packages = c("retry",
                             "zip",
                             'ggplot2',
                             'lubridate',
-                            'scales'))
+                            'scales',
+                            'ggforce',
+                            'showtext',
+                            'palatteer',
```
Comment (aaarcher-usgs) [inline _targets.R]:
```suggestion
                            'paletteer',
```

File: 2_process.R [hunk 2]
```diff
@@ -10,6 +11,17 @@ p2_targets <- list(
   ###### Get 1951-2020 metadata ######
   tar_target(p2_1951_2020_metadata,
              filter(p2_metadata, national_1951)),
+  tar_target(p2_casc_list,
```
Comment (aaarcher-usgs) [inline 2_process.R:14]: Thanks for putting this in here!

File: 2_process.R [hunk 3]
```diff
@@ -21,16 +33,35 @@ p2_targets <- list(
...
+             ),
+  tar_target(p2_site_prop_10,
+             # Filter to 10 threshold
+             p2_1951_2020_drought_summary_jd %>%
+               filter(threshold == 10) %>%
+               left_join(p2_1951_2020_metadata %>%
+                           select(StaID:STATE, HCDN_2009))
+             ),
+  tar_target(p2_site_prop_2,
+             # Filter to 10 threshold
```
Comment (aaarcher-usgs) [inline 2_process.R]:
```suggestion
             # Filter to 2 threshold
```

File: 2_process/src/prep_stripswarm.R [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  n <- 600 # set arbitrarily large number of possible simultaneous drought events positions. Trimmed prior to plotting
+  
+  E <- as_tibble(matrix(NaN,nrow=n,ncol=max(event_subset$end_day)+1))
+  E <- E %>% mutate(priority = 1:n)
+  E <- E %>% arrange(desc(priority)) %>% 
+    bind_rows(E) %>% 
+    mutate(rnum = 1:(2*n))
+  
+  # loop through each event and place into best available spot in grid
+  progress_bar <- txtProgressBar(min = 1, max = nrow(event_subset), style = 3)
```
Comment (aaarcher-usgs) [inline 2_process/src/prep_stripswarm.R:19]: NEAT!

**Review by aaarcher-usgs — APPROVED** [review]
Took 8 minutes to run on my Mac - produced this figure:
swarm_test

That figure looks different than yours, slightly. Not sure if that means something didn't run correctly or if you had changed things since you pasted yours in above. 

Minor edits to make, otherwise approved.

**Comment by cnell-usgs** [comment]
Good catch, I hadn't committed the legend repositioning, added now.

**Pull request merged**